src/value_generators/test/DocumentGeneratorTestCases.yml

```diff
@@ -392,3 +392,50 @@ Tests:
           - ^RandomInt: {min: 10000000050, max: 10000000060}
           - ^RandomString: {length: 10000000015}
           - scalarValue
+
+  - Name: ChooseGenerator
+    GivenTemplate:
+      a: {^Choose: {from: [1, "abc", {a: 1}]}}
+    ThenReturns:
+      - a: abc
+      - a: abc
+      - a: {a: 1}
+      - a: 1
+
+  - Name: Join Generator
+    GivenTemplate:
+      a: {^Join: {array: ["It", "worked"]}}
+    ThenReturns:
+      - a: "Itworked"
+
+  - Name: Join Generator with seperator
+    GivenTemplate:
+      a: {^Join: {array: ["It", "worked"], sep: " "}}
+    ThenReturns:
+      - a: "It worked"
+
+  - Name: Join With Choose
+    GivenTemplate:
+      a: {^Join: {array: ["Hello", "random", "person:", {^Choose: {from: ["Huey", "Duey",
+    {^FastRandomString: {length: 10}}]}}], sep: " "}}
+    ThenReturns:
+      - a: "Hello random person: Duey"
+      - a: "Hello random person: Duey"
+      - a: "Hello random person: smlCLkHjFD"
+      - a: "Hello random person: mnqojklKFs"
+
+  - Name: Double Generator
```

Comment:
I didn't test all the distributions. I'm not sure I want to, but I can if you would like.

Reply:
> I didn't test all the distributions.

I think just a single test that exercises all distributions would be worthwhile to ensure we don't break something in the future.

```
GivenTemplate:
  laplace: {^RandomDouble: {withDistribution: laplace ...}
  chi_squared: ....
```

(syntax is wrong but hopefully that's clear enough)